qos-service-impl/src/main/java/com/palantir/atlasdb/qos/ratelimit/QosRateLimiter.java

```diff
@@ -123,9 +123,14 @@ private synchronized void createRateLimiterAtomically() {
      */
     public void recordAdjustment(long adjustmentUnits) {
         if (adjustmentUnits > 0) {
-            rateLimiter.steal(Ints.saturatedCast(adjustmentUnits)); // TODO(nziebart): deal with longs
+            rateLimiter.steal(adjustmentUnits);
+        } else {
+            try {
+                rateLimiter.returnPermits(LongMath.checkedMultiply(adjustmentUnits, -1), stopwatch.readNanos());
```

Comment:
nit: let's name this `return` for consistency with the other method names

Reply:
LOL, `return` is a keyword.